packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx

```diff
@@ -41,11 +41,20 @@ const Zone = ({
       // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
       ([, { value }]) => value.type === "UID",
     );
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+    const widgets = widgetsArray.filter(({ TYPE_NAME }) => TYPE_NAME !== "UID");
+
     // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-    return hasUid
-      ? // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
-        widgetsArray.filter(({ TYPE_NAME }) => TYPE_NAME !== "UID")
-      : widgetsArray;
+    if (hasUid) return widgets;
+
+    // Move UID widget to the end of the array
```

Comment:
⚠️ This comment explains what the code does (which in this case it's self-explanatory) but not why we do it, could you add a bit of context here?

Reply:
added reason